### PR TITLE
[PM-3349] [PM-3350] MAUI Migration Initial

### PR DESCRIPTION
**Pull request by fedemkr**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [X] Other

## Objective

Migrate app to NET MAUI

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Tried building this locally (on .NET SDK 8.0.0) and it fails with:

```
/Users/bruno/.nuget/packages/microsoft.maui.resizetizer/8.0.3/buildTransitive/Microsoft.Maui.Resizetizer.After.targets(619,9): error : One or more duplicate file names were detected.  All image output filenames must be unique: dotnet_bot (/Users/bruno/git/bitwarten-mobile/src/Core/Resources/Images/dotnet_bot.svg), dotnet_bot (/Users/bruno/git/bitwarten-mobile/src/Core/Resources/Images/dotnet_bot.svg), dotnet_bot (/Users/bruno/git/bitwarten-mobile/src/Core/Resources/Images/dotnet_bot.svg) [/Users/bruno/git/bitwarten-mobile/src/App/App.csproj::TargetFramework=net8.0-ios]
    995 Warning(s)
    1 Error(s)
```

But I see a single `dotnet_bot.svg` so not sure what's going on.

Y'all can build the last commit? (CI is broken)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -466,21 +483,38 @@ jobs:
...
-    runs-on: macos-12
+    runs-on: macos-13
     needs: setup
+    env:
+      ios_folder_path: src/App/Platforms/iOS
+      app_output_name: App
+      app_ci_output_filename: App_x64_Debug
     steps:
+      - name: Set XCode version
+        uses: maxim-lobanov/setup-xcode@v1
```
Comment (vgrassia) [inline .github/workflows/build.yml:496]:
```suggestion
        uses: maxim-lobanov/setup-xcode@60606e260d2fc5762a71e64e74b2174e8ea3c8bd # v1.6.0
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

File: src/Core/Pages/Vault/CipherAddEditPage.xaml [hunk 2]
```diff
@@ -129,10 +129,9 @@
                         StyleClass="box-label" />
                     <Entry
                         x:Name="_nameEntry"
-                        Text="{Binding Cipher.Name}"
+                        Text="{Binding Cipher.Name, FallbackValue=''}"
                         StyleClass="box-value"
-                        AutomationProperties.IsInAccessibleTree="True"
-                        AutomationProperties.Name="{u:I18n Name}"
+                        SemanticProperties.Description="{u:I18n Name}"
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/Pages/Vault/CipherAddEditPage.xaml:134]: If we proceed with this change there a couple more places that this is happening and needs to me fixed.

**Review by andrebispo5 — COMMENTED** [review]
Awesome work! 
Just a couple of things that came to attention on a first pass.

File: src/App/Platforms/Android/Handlers/LabelHandlerMappings.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                }
+            });
+
+            Microsoft.Maui.Handlers.LabelHandler.Mapper.AppendToMapping(nameof(ILabel.AutomationId), (handler, label) =>
+            {
+                handler.PlatformView.ContentDescription = label.AutomationId;
+            });
+
+            // WORKAROUND: There is an issue causing Multiline Labels that also have a LineBreakMode to not display text properly. (it truncates text on first line even with space available)
+            // MAUI Github Issue: https://github.com/dotnet/maui/issues/14125 and https://github.com/dotnet/maui/pull/14918
```
Comment (aj-rosado) [inline src/App/Platforms/Android/Handlers/LabelHandlerMappings.cs:32]: This issue has been recently fixed.

File: src/App/Platforms/Android/MainActivity.cs [hunk 4]
```diff
@@ -136,7 +134,7 @@ protected override void OnPause()
         protected override void OnResume()
         {
             base.OnResume();
-            Xamarin.Essentials.Platform.OnResume();
+            //Xamarin.Essentials.Platform.OnResume();
```
Comment (aj-rosado) [inline src/App/Platforms/Android/MainActivity.cs:137]: If this is not necessary anymore we could remove this

File: src/App/Platforms/Android/MainActivity.cs [hunk 5]
```diff
@@ -224,8 +222,9 @@ protected override void OnNewIntent(Intent intent)
             }
             else
             {
-                Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);
-                PermissionsHandler.OnRequestPermissionsResult(requestCode, permissions, grantResults);
+                Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);
+                //Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);
```
Comment (aj-rosado) [inline src/App/Platforms/Android/MainActivity.cs:226]: If not necessary, remove the commented code

File: src/App/Platforms/Android/MainApplication.cs [hunk 6]
```diff
@@ -122,22 +123,22 @@ private void RegisterLocalServices()
...
-                FFImageLoading.ImageService.Instance.Initialize(new FFImageLoading.Config.Configuration
-                {
-                    FadeAnimationEnabled = false,
-                    FadeAnimationForCachedImages = false,
-                    HttpClient = new HttpClient(new AndroidClientHandler() { AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate })
-                });
-                ZXing.Net.Mobile.Forms.Android.Platform.Init();
-            });
-            CrossFingerprint.SetCurrentActivityResolver(() => CrossCurrentActivity.Current.Activity);
+            //Task.Run(() =>
```
Comment (aj-rosado) [inline src/App/Platforms/Android/MainApplication.cs:126]: Can this be removed?

File: src/Core/Controls/Avatar/AvatarImageSource.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            _text = name;
+            if (string.IsNullOrWhiteSpace(_text))
+            {
+                _text = email;
+            }
+            _color = color;
+
+            //Workaround: [MAUI-Migration] There is currently a bug in MAUI where the actual size of the image is used instead of the size it should occupy in the Toolbar.
+            //This causes some issues with the position of the icon. As a workaround we make the icon smaller until this is fixed.
+            //Github issues: https://github.com/dotnet/maui/issues/12359  and  https://github.com/dotnet/maui/pull/17120
```
Comment (aj-rosado) [inline src/Core/Controls/Avatar/AvatarImageSource.cs:41]: This issues have been fixed

**Review by aj-rosado — COMMENTED** [review]
Great work! 
Just added some comments regarding cleanup work and a couple workarounds that seem to be fixed.

File: src/Core/Controls/AuthenticatorViewCell/AuthenticatorViewCell.xaml [hunk 8]
```diff
@@ -43,13 +39,15 @@
         IsVisible="{Binding ShowIconImage}"
         Source="{Binding IconImageSource, Mode=OneTime}"
         AutomationProperties.IsInAccessibleTree="False" />
+        <!-- ErrorPlaceholder="login.png"
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply src/Core/Controls/AuthenticatorViewCell/AuthenticatorViewCell.xaml:42]: Yep, they are going to be removed in #2969

File: src/Core/Controls/IconButton.cs [hunk 9]
```diff
@@ -8,13 +9,14 @@ public class IconButton : Button
         public IconButton()
         {
             Padding = 0;
+            // TODO Xamarin.Forms.Device.RuntimePlatform is no longer supported. Use Microsoft.Maui.Devices.DeviceInfo.Platform instead. For more details see https://learn.microsoft.com/en-us/dotnet/maui/migration/forms-projects#device-changes
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply src/Core/Controls/IconButton.cs:12]: As you can see `Device.RuntimePlatform` is still being used, it should be changed for `DeviceInfo.Platform` instead or use precompiler directives. However, this is really low priority.

File: src/Core/Controls/MiButton.cs [hunk 10]
```diff
@@ -7,13 +8,14 @@ public class MiButton : Button
         public MiButton()
         {
             Padding = 0;
+            // TODO Xamarin.Forms.Device.RuntimePlatform is no longer supported. Use Microsoft.Maui.Devices.DeviceInfo.Platform instead. For more details see https://learn.microsoft.com/en-us/dotnet/maui/migration/forms-projects#device-changes
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply src/Core/Controls/MiButton.cs:11]: As you can see `Device.RuntimePlatform` is still being used, it should be changed for `DeviceInfo.Platform` instead or use precompiler directives. However, this is really low priority.

File: src/Core/Controls/MiLabel.cs [hunk 11]
```diff
@@ -1,18 +1,20 @@
...
+﻿using Microsoft.Maui.Controls;
+using Microsoft.Maui;
 
 namespace Bit.App.Controls
 {
     public class MiLabel : Label
     {
         public MiLabel()
         {
+            // TODO Xamarin.Forms.Device.RuntimePlatform is no longer supported. Use Microsoft.Maui.Devices.DeviceInfo.Platform instead. For more details see https://learn.microsoft.com/en-us/dotnet/maui/migration/forms-projects#device-changes
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply src/Core/Controls/MiLabel.cs:10]: As you can see `Device.RuntimePlatform` is still being used, it should be changed for `DeviceInfo.Platform` instead or use precompiler directives. However, this is really low priority.

File: src/Core/Pages/Vault/CipherAddEditPage.xaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fedemkr) [reply src/Core/Pages/Vault/CipherAddEditPage.xaml:134]: Nice finding 🎉 , can you create a Jira ticket with this? At the moment, we just replaced the whole solution `Name` with `Description` to compile, but yes we should definitely tackle this at some point.

File: src/App/Platforms/Android/Handlers/LabelHandlerMappings.cs (hunk 3 above)
Comment (fedemkr) [reply src/App/Platforms/Android/Handlers/LabelHandlerMappings.cs:32]: That is marked as it's going to be released in SR2 and it's not out yet.

File: src/Core/Controls/Avatar/AvatarImageSource.cs (hunk 7 above)
Comment (fedemkr) [reply src/Core/Controls/Avatar/AvatarImageSource.cs:41]: I think @dinisvieira tested it and the issue was still happening. Was it? Maybe we should check again.

File: src/App/Platforms/Android/Handlers/LabelHandlerMappings.cs (hunk 3 above)
Comment (fedemkr) [reply src/App/Platforms/Android/Handlers/LabelHandlerMappings.cs:32]: I thought there was one, but if not yes if you can create it please 🎉

**Review by LRNcardozoWDF — DISMISSED** [review]
Amazing work! Just found a nitpick.

**Review by aj-rosado — APPROVED** [review]
LGTM!

**Review by vgrassia — APPROVED** [review]

**Review by mpbw2 — APPROVED** [review]

**Pull request merged**